### PR TITLE
[doc] generate documentation

### PR DESCRIPTION
**Pull request by bukepo**
This commit automatically publishes the generated documentation to `gh-pages` branch.

### CODE REVIEW

**Comment by bukepo** [comment]
Preview https://bukepo.github.io/ot-br-posix/

**Review by gjc13 — APPROVED** [review]
Cool, thanks!

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+##  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+##  POSSIBILITY OF SUCH DAMAGE.
+##
+
+name: Documentation
+
+on:
+  push:
+    branches:
+      - master
```
Comment (jwhui) [inline .github/workflows/documentation.yml:34]:
```suggestion
    - master
```

File: doc/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    configure_file(${DOXYGEN_IN} ${DOXYGEN_OUT} @ONLY)
+    message("Doxygen build started")
+
+    add_custom_target(doxygen ALL
+        COMMAND ${DOXYGEN_EXECUTABLE} ${DOXYGEN_OUT}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+        COMMENT "Generating documentation with Doxygen"
+        VERBATIM)
+else()
+    message("Doxygen need to be installed to generate the doxygen documentation")
```
Comment (jwhui) [inline doc/CMakeLists.txt:44]:
```suggestion
    message("Doxygen must be installed to generate doxygen documentation")
```

**Review by jwhui — APPROVED** [review]

**Pull request merged**